street_network/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,128 @@
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
             changes.push((*id, guess_complexity(streets, id)));
         }
     }
 
     for (id, complexity) in changes.into_iter() {
-        streets.intersections.get_mut(&id).unwrap().complexity = complexity;
+        let intersection = streets.intersections.get_mut(&id).unwrap();
+        intersection.complexity = complexity.0;
+        intersection.conflict_level = complexity.1;
     }
 }
 
 /// Guesses the complexity of the intersection based on the connecting roads and their lanes.
 ///
 /// The existing complexity field is ignored, so be careful how you use the guessed value.
-fn guess_complexity(streets: &StreetNetwork, intersection_id: &NodeID) -> IntersectionComplexity {
-    let roads = streets.roads_per_intersection(*intersection_id);
+fn guess_complexity(
+    streets: &StreetNetwork,
+    intersection_id: &NodeID,
+) -> (IntersectionComplexity, ConflictType) {
+    use ConflictType::*;
+    let road_ids = streets.roads_per_intersection(*intersection_id);
+    let roads: Vec<&Road> = road_ids
+        .iter()
+        .map(|id| streets.roads.get(id).unwrap())
+        .collect();
 
     // A terminus is characterised by a single connected road.
-    if roads.len() == 1 {
-        return Terminus;
+    if road_ids.len() == 1 {
+        return (Terminus, Uncontested);
     }
 
     // A Connection is characterised by exactly two connected roads.
-    if roads.len() == 2 {
-        return Connection;
+    if road_ids.len() == 2 {
+        return (Connection, Uncontested);
     }
 
-    // A MultiConnection is characterised by exactly one dividing line traveling through it (the
-    // line that separates traffic in different directions) and no traffic that crosses it.
-    if roads.len() == 3 {
-        let mut _num_roads_in = 0;
-        let mut _num_roads_out = 0;
-        let mut num_roads_inout = 0;
-        for road_id in roads {
-            let is_outward = road_id.i1 == *intersection_id;
-            let road = streets.roads.get(&road_id).unwrap();
-            match road.oneway_for_driving() {
-                Some(Direction::Fwd) => {
-                    if is_outward {
-                        _num_roads_out += 1
-                    } else {
-                        _num_roads_in += 1
-                    }
-                }
-
-                Some(Direction::Back) => {
-                    if is_outward {
-                        _num_roads_in += 1
-                    } else {
-                        _num_roads_out += 1
-                    }
-                }
-                None => num_roads_inout += 1,
+    // Calculate all the possible movements, except (U-turns).
+    //FIXME assert!(roads is sorted clockwise), which it isn't
+    let mut connections = Vec::new();
+    // Consider turns pairs of roads, from s to d, using their position as index, so we can them later.
+    for s in 0..road_ids.len() {
```

Comment:
It might be easier to just store `&Road` references instead of the indexing and separate `road_ids` / `roads`. This whole method is `&StreetNetwork` read-only, so it's impossible for a pointer to change

Reply:
(Never mind, ordering around the intersection is why the indices are important)

Reply:
Yes, the positions are important. The road_ids / road distinction is because a `Road` doesn't know it's own id.